pyat/at/errors/error_def.py

```diff
@@ -0,0 +1,449 @@
+        dynamicerr = get_err(e, pdynamic, index)
+        pn = np.zeros(max(len(value), len(staticerr), len(dynamicerr)))
+        pn[:len(value)] += value
+        pn[:len(staticerr)] += staticerr
+        pn[:len(dynamicerr)] += e.strength*dynamicerr
+        return pn
+
+    def set_polerr(ring, pname, index):
+        pstat = pname+'Err'
+        pdyn = 'Scaling' + pstat
+        refpts = [hasattr(e, pname) and (hasattr(e, pstat) or
+                                         hasattr(e, pdyn)) for e in ring]
+        rr = ring.replace(refpts)
+        for e in rr[refpts]:
+            setattr(e, pname, get_pol(e, pname, pstat, pdyn, index))
+            sanitize(e)
+        return rr
+
+    alldef = kwargs.pop('all', True)
+    for pol in ['A', 'B']:
+        if kwargs.pop('Polynom'+pol, alldef):
+            index = kwargs.pop('Index'+pol, None)
+            ring = set_polerr(ring, 'Polynom'+pol, index)
+    return ring
+    
+
+def enable_errors(ring: Lattice, **kwargs):
+    r"""Enable magnet errors
+
+    Returns a shallow copy of *ring* where magnet errors defined by
+    :py:func:`assign_errors` are introduced in the magnet attributes.
+
+    By default, all assigned errors are enabled, but keyword arguments allow a
+    selection of the enabled errors.
+
+    Args:
+        ring:   Lattice description.
+
+    Keyword Args:
+        all (bool):         Set the default value for all the specific error
+          flags. Default: :py:obj:`True`
+        shiftx (bool):      enable horizontal shift errors. Default: *all*
+        shifty (bool):      enable vertical shift errors. Default: *all*
+        tilt (bool):        enable tilt errors. Default: *all*
+        pitch (bool):       enable pitch errors. Default: *all*
+        yaw (bool):         enable yaw errors. Default: *all*
+        PolynomB (bool):    enable polynomial errors. Default: *all*
+        IndexB (Optional[int]): restrict the polynomial errors to the
+          specified index. Default: :py:obj:`None` meaning the whole polynom
+        PolynomA (bool):    enable polynomial errors. Default: *all*
+        IndexA (Optional[int]): restrict the skew polynomial error to the
+          specified index. Default: :py:obj:`None` meaning the whole polynom
+
+    Examples:
+        >>> ringerr = enable_errors(ring)
+
+        All errors are enabled in *ringerr*
+
+        >>> ringerr = enable_errors(all=False, ShiftErr=True)
+
+        Only magnet shift errors are active
+
+        >>> ringerr = enable_errors(all=False)
+
+        No magnet errors are enabled. Only monitor errors will be active.
+
+    See also:
+        :py:func:`assign_errors`, :py:func:`get_optics_err`
+    """
+    if not ring.has_errors:
+        ring = _apply_field_errors(ring, **kwargs)
+        ring = _apply_alignment_errors(ring, **kwargs)
+        ring._has_errors = True
+    return ring
+
+
+def has_errors(ring):
+    """Get the error status of a lattice"""
+    return getattr(ring, '_has_errors', False)
+
```

Comment:
an additional function to print a text table of errors is often really useful to read a lattice

Reply:
Good point. I take note

Reply:
@simoneliuzzo we had been discussing the use of ``pandas`` for this but I did not implement it since you were interested to look at it yourself, is it still something you intend to do?

Reply:
@simoneliuzzo and @swhite2401 : I hope you don't intend to have PyAT depend on `pandas`…

Reply:
Well I do not see a real need for now, but `pandas` offers a lots of functionalities that could become handy and is sufficiently well documented and widely used that it could be considered.
Why do you see this as a problem? How is it more of a problem than `mpi4py` for instance?